### DISCUSSION TITLE
Provide a way to register a custom Projection ol.proj.proj4

### DISCUSSION THREAD

**Discussion by eltorio** [MelihAltintas/vue3-openlayers#58] [Q&A] [answered]
Dear Melih,

It seems to be a good improvement to provide a way to register new projection.
Only EPSG:3857 and EPSG:4326 are included in openlayers 6,
Creating a projection with ol.proj.Projection(…) is not sufficient because there is so much parameters.
For example in France we use RGF93  (epsg:2154) coordinates so we need to define:

```
//Proj4 initialization
import proj4 from 'proj4';
import {register} from 'ol/proj/proj4';
proj4.defs("EPSG:2154", "+proj=lcc +lat_1=49 +lat_2=44 +lat_0=46.5 +lon_0=3 +x_0=700000 +y_0=6600000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs");
register(proj4);
//if needed
const epsg2154_fullextent = [-357823.2365, 6037008.6939, 1313632.3628, 7230727.3772];
```

Thanks for your work !

**Comment by MelihAltintas** [comment, marked as answer]
thanks for your suggestion.

I added a new component for registration. You can easily register the new project with:

```
    <ol-projection-register projectionName="EPSG:21781"
    :projectionExtent="[-357823.2365, 6037008.6939, 1313632.3628, 7230727.3772]"
    projectionDef="+proj=lcc +lat_1=49 +lat_2=44 +lat_0=46.5 +lon_0=3 +x_0=700000 +y_0=6600000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs"/>
```

then you can use like that:

` <ol-view ref="view" :center="center" :rotation="rotation" :zoom="zoom" projection="EPSG:21781" />`

please update your version to V.0.1.56

example: https://codesandbox.io/s/projection-register-example-kl0qs?file=/src/App.vue

**Comment by eltorio** [comment]
Thanks a lot…
It works like a charm !
